### PR TITLE
feat(entoas): added field annotation for skip flag

### PR DESCRIPTION
**Pull request by badygin-yakov**
Added Ignored field to Annotation struct to support ignored filed in spec (useful for deletedAt field or system fields don't need in spec).

### CODE REVIEW

**Comment by a8m** [comment]
cc @masseelch.

Also, please use Skip instead of Ignored to keep it consistent with the rest of the packages here (e.g. entgql).

**Comment by badygin-yakov** [comment]
@a8m I can rename field or need realize logic [SkipMode](https://github.com/ent/contrib/blob/78aa88292a6251eaaae6d5c79dd41a7124c41006/entgql/annotation.go#L83-L107)?

**Comment by badygin-yakov** [comment]
@masseelch review pls

**Comment by masseelch** [comment]
@badygin-yakov I am not sure what this PR does. For what I can see it adds an annotation but does not do anything with it. Can you explain?

File: entoas/generator.go [hunk 1]
```diff
@@ -541,6 +548,11 @@ func OgenSchema(f *gen.Field) (*ogen.Schema, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if ant.Skip {
+		return nil, nil
+	}
+
```
Comment (masseelch) [inline entoas/generator.go:555]: I prefer to have this done more explicit by either removing the field [here](https://github.com/ent/contrib/blob/e738ef866e51c76d31fcc71919a7986d2e403719/entoas/generator.go#L59) or maybe [here](https://github.com/ent/contrib/blob/e738ef866e51c76d31fcc71919a7986d2e403719/entoas/generator.go#L127)

**Review by masseelch — CHANGES_REQUESTED** [review]
Hey @badygin-yakov, thanks for this PR.

Please see my comment.

File: entoas/generator.go (hunk 1 above)
Comment (badygin-yakov) [reply entoas/generator.go]: Okey, I refactored code and made skip logic more explicit

File: entoas/generator.go [hunk 2]
```diff
@@ -126,6 +126,15 @@ func schemas(g *gen.Graph, spec *ogen.Spec) error {
 // addSchemaFields adds the given gen.Field slice to the ogen.Schema.
 func addSchemaFields(s *ogen.Schema, fs []*gen.Field) error {
 	for _, f := range fs {
+		ant, err := FieldAnnotation(f)
+		if err != nil {
+			return err
+		}
+
```
Comment (masseelch) [inline entoas/generator.go:133]:
```suggestion
```

File: entoas/generator.go [hunk 3]
```diff
@@ -126,6 +126,15 @@ func schemas(g *gen.Graph, spec *ogen.Spec) error {
...
+		if ant.Skip {
+			continue
+		}
+
```
Comment (masseelch) [inline entoas/generator.go:137]:
```suggestion
```

File: entoas/generator.go [hunk 4]
```diff
@@ -678,7 +687,12 @@ func reqBody(n *gen.Type, op Operation) (*ogen.RequestBody, error) {
 		if err != nil {
 			return nil, err
 		}
-		if (a != nil && !a.ReadOnly) && (op == OpCreate || !f.Immutable) {
+
```
Comment (masseelch) [inline entoas/generator.go:690]:
```suggestion
```

File: entoas/generator.go [hunk 5]
```diff
@@ -678,7 +687,12 @@ func reqBody(n *gen.Type, op Operation) (*ogen.RequestBody, error) {
...
+		if a.ReadOnly || a.Skip {
+			continue
+		}
+
```
Comment (masseelch) [inline entoas/generator.go:694]:
```suggestion
```

File: entoas/generator.go [hunk 6]
```diff
@@ -678,7 +687,12 @@ func reqBody(n *gen.Type, op Operation) (*ogen.RequestBody, error) {
 		if err != nil {
 			return nil, err
 		}
-		if (a != nil && !a.ReadOnly) && (op == OpCreate || !f.Immutable) {
+
+		if a.ReadOnly || a.Skip {
```
Comment (masseelch) [inline entoas/generator.go:691]: The check for `a != nil` is required, isn't it?

File: entoas/internal/simple/schema/category.go [hunk 7]
```diff
@@ -15,10 +15,11 @@
 package schema
 
 import (
-	"entgo.io/contrib/entoas"
 	"entgo.io/ent"
 	"entgo.io/ent/schema/edge"
 	"entgo.io/ent/schema/field"
+
```
Comment (masseelch) [inline entoas/internal/simple/schema/category.go:21]:
```suggestion
```

**Review by masseelch — CHANGES_REQUESTED** [review]
Nice! See some suggestions, then I can merge :-)

File: entoas/generator.go (hunk 6 above)
Comment (badygin-yakov) [reply entoas/generator.go:688]: Why need check? `a` can be nil only if `FieldAnnotation` return err. But this case we handle on [684-687](https://github.com/ent/contrib/pull/329/files#diff-8a343f04c5df9bc0b27ce52ca7989745e813d4779329f49f69ca18d74963c993R684-R687)

**Review by masseelch — APPROVED** [review]

**Comment by masseelch** [comment]
Please fix the CI, then I will merge

**Pull request merged**